smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/commit/PartitionTest.java

```diff
@@ -0,0 +1,183 @@
+import static org.awaitility.Awaitility.await;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ConsumerGroupListing;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.reactive.messaging.kafka.KafkaConnector;
+import io.smallrye.reactive.messaging.kafka.base.KafkaTestBase;
+import io.smallrye.reactive.messaging.test.common.config.MapBasedConfig;
+
+public class PartitionTest extends KafkaTestBase {
+
+    @Test
+    public void testWithPartitions() throws ExecutionException, InterruptedException {
+        createTopic(topic, 3);
+
+        MapBasedConfig config = new MapBasedConfig()
+                .with("mp.messaging.incoming.kafka.connector", KafkaConnector.CONNECTOR_NAME)
+                .with("mp.messaging.incoming.kafka.bootstrap.servers", getBootstrapServers())
+                .with("mp.messaging.incoming.kafka.topic", topic)
+                .with("mp.messaging.incoming.kafka.partitions", 3)
+                .with("mp.messaging.incoming.kafka.auto.offset.reset", "earliest")
+                .with("mp.messaging.incoming.kafka.value.deserializer", StringDeserializer.class.getName());
+
+        MyApplication application = runApplication(config, MyApplication.class);
+
+        AtomicInteger count = new AtomicInteger();
+        int expected = 3000;
+        Random random = new Random();
+        usage.produce(topic, expected, new StringSerializer(), new StringSerializer(), null, () -> {
+            int value = count.getAndIncrement();
+            int p = random.nextInt(3);
+            return new ProducerRecord<>(topic, p, Integer.toString(p), Integer.toString(value));
+        });
+
+        await()
+                .atMost(30, TimeUnit.SECONDS)
+                .until(() -> application.count() == expected);
+        assertThat(application.getReceived().keySet()).hasSize(3);
+
+        Properties properties = new Properties();
+        properties.put("bootstrap.servers", getBootstrapServers());
+        Admin admin = Admin.create(properties);
+        ConsumerGroupListing group = admin.listConsumerGroups().all().get().stream().findFirst().orElse(null);
```

Comment:
There's multiple tests in this class that share the same Kafka broker, and the incoming channel config doesn't define a group ID, so group IDs will be autogenerated. So there will be eventually 3 consumer groups, and you don't know which one you get. No?

Reply:
OH you are absolutely right!

Reply:
Defining a unique group ID in channel config and using `Admin.listConsumerGroupOffsets(String)` should work IMHO.

Reply:
Fixed